### PR TITLE
Clean up unit testing sample using 0.6 features

### PR DESCRIPTION
**Pull request by cgranade**
This PR cleans up the unit testing sample using the new features of Q# 0.6 (e.g.: removes redundancy, makes use of improved library routines, etc.).

### CODE REVIEW

File: Samples/src/UnitTesting/CCNOTCircuits.qs [hunk 1]
```diff
@@ -296,28 +225,29 @@ namespace Microsoft.Quantum.Samples.UnitTesting {
...
         // fall back to a standard multiply controlled X Implementation
         controlled (controls, ...) {
             Controlled X(controls + [control1, control2], target);
         }
 
+        // NB: We cannot use "is Adj" or "adjoint auto" here, due to the use of
+        //     Microsoft.Quantum.Intrinsic.M, which does is not Adj. Instead, we
+        //     explicitly use the "self" generator.
+        adjoint self;
         controlled adjoint self;
```
Comment (bettinaheim) [inline Samples/src/UnitTesting/CCNOTCircuits.qs]: this line is not needed (no longer causes a warning) - either the adjoint self or controlled adjoint self declaration is sufficient

File: Samples/src/UnitTesting/CCNOTCircuits.qs [hunk 2]
```diff
@@ -331,30 +261,23 @@ namespace Microsoft.Quantum.Samples.UnitTesting {
...
-            ExpFrac([PauliZ, PauliI, PauliI], -1, 3, register);
-            ExpFrac([PauliI, PauliZ, PauliI], -1, 3, register);
-            ExpFrac([PauliI, PauliI, PauliZ], -1, 3, register);
-            ExpFrac([PauliZ, PauliZ, PauliI], 1, 3, register);
-            ExpFrac([PauliZ, PauliI, PauliZ], 1, 3, register);
-            ExpFrac([PauliI, PauliZ, PauliZ], 1, 3, register);
-            ExpFrac([PauliZ, PauliZ, PauliZ], -1, 3, register);
-        }
-
-        adjoint self;
```
Comment (bettinaheim) [inline Samples/src/UnitTesting/CCNOTCircuits.qs]: self should be preserved for optimization purposes (it's tracked as inferred property in the compiler)

File: Samples/src/UnitTesting/CCNOTCircuits.qs [hunk 3]
```diff
@@ -148,30 +112,23 @@ namespace Microsoft.Quantum.Samples.UnitTesting {
...
-            Adjoint T(control1);
-            CNOT(control2, control1);
-            H(target);
-            Adjoint T(control1);
-            CNOT(control2, control1);
-            T(control2);
-            S(control1);
-        }
-
-        adjoint self;
```
Comment (bettinaheim) [inline Samples/src/UnitTesting/CCNOTCircuits.qs]: self should be preserved for optimization purposes (it's tracked as inferred property in the compiler)

File: Samples/src/UnitTesting/CCNOTCircuits.qs [hunk 4]
```diff
@@ -187,29 +144,22 @@ namespace Microsoft.Quantum.Samples.UnitTesting {
...
-            Adjoint T(control1);
-            CNOT(control2, target);
-            CNOT(target, control1);
-            Adjoint T(target);
-            T(control1);
-            H(target);
-            CNOT(control2, control1);
-        }
-
-        adjoint self;
```
Comment (bettinaheim) [inline Samples/src/UnitTesting/CCNOTCircuits.qs]: same here

File: Samples/src/UnitTesting/CCNOTCircuits.qs [hunk 5]
```diff
@@ -363,17 +286,10 @@ namespace Microsoft.Quantum.Samples.UnitTesting {
...
     /// Uses 7 T gates, 10 CNOTs and two Hadamard gates and has T depth 5
-    operation CCNOT4 (control1 : Qubit, control2 : Qubit, target : Qubit) : Unit {
-
-        body (...) {
-            H(target);
-            CCZ1(control1, control2, target);
-            H(target);
-        }
-
-        adjoint self;
```
Comment (bettinaheim) [inline Samples/src/UnitTesting/CCNOTCircuits.qs]: same here (keep adjoint self)

File: Samples/src/UnitTesting/ControlledSWAP.qs [hunk 6]
```diff
@@ -1,63 +1,49 @@
...
-        body (...) {
-            // Note that SWAP(a,b) = CNOT(b,a) CNOT(a,b) CNOT(b,a)
-            // Since CNOT(b,a) is self-adjoint: CNOT(b,a)CNOT(b,a)=I,
-            // Controlled SWAP(a,b) = CNOT(b,a) CCNOT(c,a,b) CNOT(b,a)
-            CNOT(target2, target1);
-            CCNOTOp(control, target1, target2);
-            CNOT(target2, target1);
-        }
-        
-        adjoint self;
```
Comment (bettinaheim) [inline Samples/src/UnitTesting/ControlledSWAP.qs]: ditto

File: Samples/src/UnitTesting/CCNOTCircuits.qs (hunk 3 above)
Comment (cgranade) [reply Samples/src/UnitTesting/CCNOTCircuits.qs]: Reverted.

File: Samples/src/UnitTesting/CCNOTCircuits.qs (hunk 4 above)
Comment (cgranade) [reply Samples/src/UnitTesting/CCNOTCircuits.qs]: Reverted.

**Pull request merged**